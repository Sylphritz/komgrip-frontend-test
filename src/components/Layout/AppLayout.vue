<template>
  <LayoutSidebar />
  <main class="main-content position-relative max-height-vh-100 h-100 mt-1 border-radius-lg">
    <NavBar />
    <LayoutBody>
      <slot />
    </LayoutBody>
  </main>
</template>

<script setup lang="ts">
import LayoutSidebar from '@/components/Layout/LayoutSidebar.vue'
import NavBar from './NavBar.vue'
import LayoutBody from './LayoutBody.vue'
</script>
