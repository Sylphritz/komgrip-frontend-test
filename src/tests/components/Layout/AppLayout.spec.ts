import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import AppLayout from '@/components/Layout/AppLayout.vue'

describe('AppLayout', () => {
  it('renders successfully and matches the snapshot', () => {
    const wrapper = mount(AppLayout, {
      shallow: true,
      slots: {
        default: 'testbody'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
