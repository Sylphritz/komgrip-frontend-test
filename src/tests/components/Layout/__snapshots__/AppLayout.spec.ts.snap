// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppLayout > renders successfully and matches the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <layout-sidebar-stub />
  <main
    class="main-content position-relative max-height-vh-100 h-100 mt-1 border-radius-lg"
  >
    <nav-bar-stub />
    <layout-body-stub />
  </main>
  
</div>
`;
